### PR TITLE
Create extension to trim url

### LINKED ISSUE #12804: [Bug] TransactionTooLargeException from "Open links in apps"
Issue seen on Fenix - https://github.com/mozilla-mobile/fenix/issues/27042
Seems like [queryIntentActivities from AppLinksUseCases](https://github.com/mozilla-mobile/android-components/blob/6431cfeee9987337ff9d9260f5a7ac6188bf6160/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt#L58) can throw this error.
Could not reproduce locally, the issue depends on the specific configuration of app and other installed apps on the device - with the functionality querying which of the currently installed apps have declared components with intent filters for the current url.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21614)

### LINKED ISSUE #12930: Ensure cleanup of the tabs presenter to prevent memory leaks
As a followup to https://github.com/mozilla-mobile/fenix/issues/27374 where it was seen that it is possible to leak an entire layout through `feature.syncedtabs.presenter.DefaultPresenter` and `FxaAccountManager`.

### PR DESCRIPTION
**Pull request by sarah541**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation
Fixes #12804
Fixes #12930

### CODE REVIEW

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/Extensions.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import java.text.NumberFormat
+import java.util.Locale
+
+// This is used for truncating URLs to prevent extreme cases from
+// slowing down UI rendering e.g. in case of a bookmarklet or a data URI.
+// https://github.com/mozilla-mobile/android-components/issues/5249
+const val MAX_URI_LENGTH = 25000
+
+fun getTrimmedUrl(url: String): String {
```
Comment (gabrielluong) [inline components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/Extensions.kt:15]: I am wondering if this should be moved to one of our more generic support components since this is being used more widely outside this browser-toolbar component. One idea is support-ktx https://searchfox.org/mozilla-mobile/source/android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt

We also want to make this an actual extension of Strings 

```suggestion
fun String.toTrimmedUrl(): String {
```

Comment (gabrielluong) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/Extensions.kt:15]: Also, please add a kDoc

Comment (Amejia481) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/Extensions.kt:15]: Agree!

Comment (sarah541) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/Extensions.kt:15]: I was kinda looking for this. Thank you!

File: buildSrc/src/main/java/Gecko.kt [hunk 2]
```diff
@@ -9,7 +9,7 @@ object Gecko {
     /**
      * GeckoView Version.
      */
-    const val version = "107.0.20221011093208"
+    const val version = "107.0.20221012094509"
```
Comment (Amejia481) [inline buildSrc/src/main/java/Gecko.kt:12]: nit: Looks like it was commit by mistake :)

File: components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt [hunk 3]
```diff
@@ -51,11 +51,19 @@ class AppLinksUseCases(
     private val launchInApp: () -> Boolean = { false },
     private val alwaysDeniedSchemes: Set<String> = ALWAYS_DENY_SCHEMES
 ) {
-    @Suppress("QueryPermissionsNeeded") // We expect our browsers to have the QUERY_ALL_PACKAGES permission
+    @Suppress(
```
Comment (Amejia481) [inline components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt:54]: Nit: Looks like unrelated changes

File: components/feature/prompts/src/main/res/values-en-rCA/strings.xml [hunk 4]
```diff
@@ -81,6 +81,8 @@
     <string name="mozac_feature_prompts_nov">Nov</string>
     <!-- December month of the year (short description), used on the month chooser dialog. -->
     <string name="mozac_feature_prompts_dec">Dec</string>
+    <!-- Title of the time picker dialog. -->
+    <string name="mozac_feature_prompts_set_time">Set time</string>
```
Comment (Amejia481) [inline components/feature/prompts/src/main/res/values-en-rCA/strings.xml:85]: Same as https://github.com/mozilla-mobile/android-components/pull/12928/files#r993886594

File: components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/presenter/DefaultPresenter.kt [hunk 5]
```diff
@@ -81,7 +81,8 @@ internal class DefaultPresenter(
     }
 
     override fun stop() {
-        // no-op
+        accountManager.unregisterForSyncEvents(eventObserver)
+        accountManager.unregister(accountObserver)
```
Comment (Amejia481) [inline components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/presenter/DefaultPresenter.kt:85]: Same as https://github.com/mozilla-mobile/android-components/pull/12928/files#r993886594

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt [hunk 6]
```diff
@@ -290,3 +295,10 @@ fun String.getRepresentativeCharacter(): String {
 fun String.last4Digits(): String {
     return this.takeLast(LAST_VISIBLE_DIGITS_COUNT)
 }
+
+/**
+ * Returns truncated urls and search terms to improve performance.
+ */
+fun String.toTrimmedUrl(): String {
+    return this.take(MAX_URI_LENGTH)
+}
```
Comment (Amejia481) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt:304]: 👍🏽

**Comment by Amejia481** [comment]
I think we will need to rebase as there are some unrelated changes that have been leaked to the PR :)
